File: web/projects/portal/src/app/common/event/collect-parameters-over-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../viewsheet-client/index";
import { VariableInfo } from "../data/variable-info";

/**
 * Event used to collect filled in parameter values.
 */
export class CollectParametersOverEvent implements ViewsheetEvent {
   /**
    * Updated variables.
    */
   public variables: VariableInfo[];

   public disableAudit: boolean;
   public openVS: boolean;

   /**
    * Creates a new instance of <tt>CollectParametersOverEvent</tt>.
    *
    * @param variables the list of variables.
    */
   constructor(variables: VariableInfo[], disableAudit = false, openVS: boolean = false) {
      this.variables = variables;
      this.disableAudit = disableAudit;
      this.openVS = openVS;
   }
}

File: core/src/main/java/inetsoft/web/viewsheet/event/CollectParametersOverEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.viewsheet.event;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import inetsoft.web.composer.ws.assembly.VariableAssemblyModelInfo;
import org.immutables.value.Value;

import java.util.ArrayList;
import java.util.List;

/**
 * Class that encapsulates the parameters for setting variables.
 *
 * @since 12.3
 */
@Value.Immutable
@JsonSerialize(as = ImmutableCollectParametersOverEvent.class)
@JsonDeserialize(builder = CollectParametersOverEvent.Builder.class)
public abstract class CollectParametersOverEvent {
   @Value.Default
   public List<VariableAssemblyModelInfo> variables() {
      return new ArrayList<>();
   }

   @Value.Default
   public boolean disableAudit() {
      return false;
   }

   @Value.Default
   public boolean openVS() {
      return false;
   }

   public static Builder builder(){return new Builder();}

   public static class Builder extends ImmutableCollectParametersOverEvent.Builder {

   }
}

